tests/functionnal/RuleCommonITILObject.php

```diff
@@ -2127,4 +2135,288 @@ public function testStopProcessingAction()
+     *  the category or the description then we can read its value from $item
+     * - If the action field is more complex like an assigned group or user, we
+     *  will run a specific function supplied in $action that will fetch the
+     *  value from the database
+     *
+     * @param array  $action Action details, supplied by the data provider
+     * @param Ticket $item   Test subject
+     */
+    protected function testActionGetTestResultValue(array $action, Ticket $item)
+    {
+        if (isset($action['field_specific'])) {
+            $value = $action['field_specific']($item);
+        } else {
+            $value = $item->fields[$action['field']];
+        }
+
+        return $value;
+    }
+
+    /**
+     * Test a given ticket rule
+     *
+     * @dataprovider testActionProvider
+     *
+     * @param array  $criteria           Details of the rule criteria:
+     *                                    - condition
+     *                                    - field
+     *                                    - pattern
+     * @param array  $action             Details of the rule action:
+     *                                    - action_type
+     *                                    - field
+     *                                    - value
+     *                                    - field_specific (optionnal, will
+     *                                     contain a callback to handle fields
+     *                                     that are no part of the ticket table,
+     *                                     like assigned groups for exemple)
+     * @param string $control_test_value A control value for the criteria.
+     *                                   This value is not expected to trigger
+     *                                   the rule
+     * @param string $real_test_value    The test value value for the criteria.
+     *                                   This value is expected to trigger the
+     *                                   rule
+     * @param string $expected_value     Expected value if the test succeed.
+     *
+     * @return void
+     */
+    public function testAction(
+        array $criteria,
+        array $action,
+        string $control_test_value,
+        string $real_test_value,
+        string $expected_value
+    ): void {
+        global $DB;
+
+        $this->login();
+
+        // Disable all others rules before running the test
+        $DB->update(Rule::getTable(), ['is_active' => false], [
+            'sub_type' => "RuleAsset"
+        ]);
+        $active_rules = countElementsInTable(Rule::getTable(), [
+            'is_active' => true,
+            'sub_type'  => "RuleAsset",
+        ]);
+        $this->integer($active_rules)->isEqualTo(0);
+
+        // Create the rule
+        $rule_ticket = $this->createItem(\RuleTicket::getType(), [
+            'name'      => 'testLastInventoryUpdateCriteria',
```

Comment:
```suggestion
            'name'      => __FUNCTION__,
```

Reply:
Forgot to change this one, good catch :)